server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java

```diff
@@ -35,6 +37,8 @@
 
     Publisher<BlobId> save(BucketName bucketName, InputStream data, StoragePolicy storagePolicy);
 
+    Publisher<BlobId> save(BucketName bucketName, ByteSource data, StoragePolicy storagePolicy);
```

Comment:
Maybe we need an API equivalent to `ByteSource` of our own here (called eg `Blob`) providing content and length.

Feedback?